docs/how-to/c-functions.rst

```diff
@@ -63,34 +63,35 @@ Once you have opened the relevant header file in a text editor, you need to sear
 
     int puts(const char *);
 
-This means that ``puts`` returns an ``int`` and takes a single argument of type ``const char *`` (a C string). This translates to the following Python ``ctypes`` code:
+This means that ``puts`` returns an ``int`` and takes a single argument of type ``const char *`` (a pointer to one or more characters, i. e. a C string). This translates to the following Python ``ctypes`` code:
 
 .. code-block:: python
 
     >>> from ctypes import c_char_p, c_int
     >>> libc.puts.restype = c_int
     >>> libc.puts.argtypes = [c_char_p]
 
-Now that we have provided all of the necessary type information, we can call ``libc.puts``:
+Now that we have provided all of the necessary type information, we can call ``libc.puts``.
+
+For the ``c_char_p`` argument, we pass a byte string with the message we want to print out. ``ctypes`` automatically converts the byte string object to a ``c_char_p`` (``char *``) as the C funcion expects it. The string specifically needs to be a byte string (``bytes``), because C's ``char *`` strings are byte-based, unlike normal Python strings (``str``), which are Unicode-based.
 
 .. code-block:: python
 
-    >>> libc.puts(b"Hello!")
+    >>> res = libc.puts(b"Hello!")
     Hello!
-    10
 
 .. note::
 
-    The ``10`` is the return value of ``puts``, which is printed out by the Python REPL; you can ignore it. (A non-negative return value from ``puts`` indicates that the call was successful.)
+    If you're running this code from an editor or IDE and don't see ``Hello!`` printed out, try running the code from a Python REPL in a terminal window instead. Some editors/IDEs, such as Python's IDLE, can only capture and display output produced by high-level Python functions (such as ``print``), but not output from low-level C functions (such as ``puts``).
 
-    If you're running this code from an editor or IDE and you see only the ``10`` and not the ``Hello!``, try running the code from a Python REPL in a terminal window instead. Some editors/IDEs, such as Python's IDLE, can only capture and display output produced by high-level Python functions (such as ``print``), but not output from low-level C functions (such as ``puts``).
+    The return value of ``puts`` is ignored in this example, because it isn't significant here.
```

Comment:
Failure to check success/failure return values is a common source of bugs in C code, so I don't think it's a good idea to brush over the return value as "not significant". This is doubly important because code that returns successfully with an error value *isn't* common Python practice. If we're providing a guide that will potentially empower people without C experience to invoke C methods, it seems appropriate to warn them about the land mines waiting for them. Granted, if `puts` returns a failure value, you've got *much* bigger problems to deal with, but it's a good a time as any to point out the design pattern that exists in C APIs.

Reply:
On that topic, I'm actually not quite sure how much prior knowledge of C we should assume in this guide? I think for the most part I assumed that the reader is somewhat familiar with C, but not how to interact with C from Python - so they would know about C strings, error code return values, etc., but not how to best interact with those concepts from Python.

In any case, I think it's okay to go into a bit more detail here, even for readers who already know how error handling works in C - it's quite easy to forget these checks in the context of Python where you normally have exceptions.

Reply:
Agreed that we don't need this guide to become a tutorial on writing C, but based on the people I'm seeing come to BeeWare and Rubicon, there's a fair amount of "I have grand ideas but no experience with any language other than Python (and even my Python knowledge is limited)". 

To that end, I think it's worth pointing out the things that are likely to be issues, even if we don't provide a *deep* explanation of the details. That will provide a reminder for those who have C experience, and a starting point for further research for those that don't.